pkg/cluster/denyassignment.go

```diff
@@ -17,8 +17,8 @@ func (m *manager) createOrUpdateDenyAssignment(ctx context.Context) error {
 	}
 
 	// needed for AdminUpdate so it would not block other steps
-	if m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID == "" {
-		m.log.Print("skipping createOrUpdateDenyAssignment: SPObjectID is empty")
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() && m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.SPObjectID == "" {
```

Comment:
The implication from the comment above seems to be that if you set an exclusion rule with an empty object ID it will error out and block further admin update steps. So with that in mind, I think this condition should be something like this:

```
if (cluster is WI and any platform WI has an empty object ID) or (cluster is SP and object ID is empty)
```

I can't think of an exact situation where a PWI would have an empty object ID and we'd want to do an admin update despite that, but I don't think it hurts to be safe.

What do you think? If you choose to make this change, it'd also be nice to see unit test cases that cover the different scenarios in which we skip creating/updating the deny assignment.

Reply:
If my memory is working, I had to put this in to avoid a nil pointer dereference...?

I've updated the logic to check for those conditions and included unit tests for it